task/queue/queue.go

```diff
@@ -226,8 +256,15 @@ func (q *queue) startManager(ctx context.Context) {
 			}
 
 			select {
-			case <-ctx.Done():
-				return
+			case <-ctx.Done(): // Drain and complete any interrupted tasks
```

Comment:
Instead of doing a non-blocking select until there are no new messages, consider closing the channel after the workers waitgroup is done and drain the channel with `for tsk := range completeTask` to make the block below easier to understand.

Reply:
[Realized I hadn't responded to this yet...]

Okay, this makes sense. I added it to a later PR.